Core/include/Acts/TrackFinding/CombinatorialKalmanFilter.hpp

```diff
@@ -713,217 +716,165 @@ class CombinatorialKalmanFilter {
       using PM = TrackStatePropMask;
 
       bool isSensitive = surface->associatedDetectorElement() != nullptr;
-      bool isMaterial = surface->surfaceMaterial() != nullptr;
-
-      std::size_t nBranchesOnSurface = 0;
+      bool hasMaterial = surface->surfaceMaterial() != nullptr;
+      bool isMaterial = hasMaterial && !isSensitive;
+      bool expectMeasurements = isSensitive;
 
-      if (auto [slBegin, slEnd] = m_sourceLinkAccessor(*surface);
-          slBegin != slEnd) {
-        // Screen output message
+      if (isSensitive) {
         ACTS_VERBOSE("Measurement surface " << surface->geometryId()
                                             << " detected.");
+      } else if (isMaterial) {
+        ACTS_VERBOSE("Material surface " << surface->geometryId()
+                                         << " detected.");
+      } else {
+        ACTS_VERBOSE("Passive surface " << surface->geometryId()
+                                        << " detected.");
+        return Result<void>::success();
+      }
 
-        // Transport the covariance to the surface
+      using SourceLinkRange = decltype(m_sourceLinkAccessor(*surface));
```

Comment:
We don't want to write this out at this point?

Reply:
You mean explicitly writing this type out? I think we have no access to the range type or iterator types but maybe I am wrong

Reply:
Ah right, I think I got confused.